### ISSUE TITLE
🐛 Biome encountered an unexpected error | internalError / panic

### ISSUE THREAD

**Issue by Tandashi** [biomejs/biome#2724]
### Environment information

```block
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64        
  OS:                           windows       

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.0"    
  JS_RUNTIME_NAME:              "node"        
  NODE_PACKAGE_MANAGER:         "pnpm/9.0.6"  

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

```bash
> pnpm biome check --apply .

Source Location: crates\biome_js_semantic\src\semantic_model\scope.rs:115:33
Thread Name: biome::worker_9
Message: no entry found for key

.\src\option.ts internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × processing panicked: no entry found for key
```

```typescript
// src/option.ts
import type * as tslog from 'tslog';

export type OptionResultSuccess<R> = [R] extends [never]
  ? { type: 'success' }
  : { type: 'success'; result: R };

export type OptionResultError<E extends string = never> = {
  type: 'error';
  error: E | 'unknown';
  message: string;
};

export type OptionResult<R, E extends string = never> =
  | OptionResultSuccess<R>
  | OptionResultError<E>;

export const wrapWithOption = async <R, E extends string = never>(
  func: () => Promise<OptionResult<R, E>> | OptionResult<R, E>,
  logger: tslog.Logger<unknown>,
): Promise<OptionResult<R, E>> => {
  try {
    return await func();
  } catch (e) {
    logger.error(e);

    if (e instanceof Error) {
      return { type: 'error', error: 'unknown', message: e.message };
    }

    if (typeof e === 'string') {
      return { type: 'error', error: 'unknown', message: e };
    }

    return { type: 'error', error: 'unknown', message: 'Unknown error' };
  }
};

export const allOptionsSuccessful = <R, E extends string>(
  options: OptionResult<R, E>[],
): options is OptionResultSuccess<R>[] =>
  options.every((option) => option.type === 'success');
```

### Expected result

-

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Tandashi** [comment]
Duplicate of https://github.com/biomejs/biome/issues/2683, https://github.com/biomejs/biome/issues/2680 and https://github.com/biomejs/biome/issues/2659.
